include/nbl/builtin/hlsl/emulated_float64_t.hlsl

```diff
@@ -0,0 +1,596 @@
+            z1 = z1more2.x;
+            more2 = z1more2.y;
+            nbl::hlsl::uint32_t2 z1z2 = add64(z1, more2, 0u, z2);
+            z1 = z1z2.x;
+            z2 = z1z2.y;
+            nbl::hlsl::uint32_t2 z0more1 = umulExtended(a0, b0);
+            z0 = z0more1.x;
+            more1 = z0more1.y;
+            nbl::hlsl::uint32_t2 z0z1 = add64(z0, more1, 0u, z1);
+            z0 = z0z1.x;
+            z1 = z0z1.y;
+            nbl::hlsl::uint32_t2 more1more2 = umulExtended(a0, b1);
+            more1 = more1more2.x;
+            more2 = more1more2.y;
+            nbl::hlsl::uint32_t2 more1z2 = add64(more1, more2, 0u, z2);
+            more1 = more1z2.x;
+            z2 = more1z2.y;
+            nbl::hlsl::uint32_t2 z0z12 = add64(z0, z1, 0u, more1);
+            z0 = z0z12.x;
+            z1 = z0z12.y;
+
+
+            nbl::hlsl::uint32_t4 output;
+            output.x = z0;
+            output.y = z1;
+            output.z = z2;
+            output.w = z3;
+            return output;
+        }
+        
+        nbl::hlsl::uint32_t3 shift64ExtraRightJamming(uint32_t a0, uint32_t a1, uint32_t a2, int count)
+        {
+           nbl::hlsl::uint32_t3 output;
+           output.x = 0u;
+           
+           int negCount = (-count) & 31;
+        
+           output.z = LERP(uint32_t(a0 != 0u), a0, count == 64);
+           output.z = LERP(output.z, a0 << negCount, count < 64);
+           output.z = LERP(output.z, a1 << negCount, count < 32);
+        
+           output.y = LERP(0u, (a0 >> (count & 31)), count < 64);
+           output.y = LERP(output.y, (a0<<negCount) | (a1>>count), count < 32);
+        
+           a2 = LERP(a2 | a1, a2, count < 32);
+           output.x = LERP(output.x, a0 >> count, count < 32);
+           output.z |= uint32_t(a2 != 0u);
+        
+           output.x = LERP(output.x, 0u, (count == 32));
+           output.y = LERP(output.y, a0, (count == 32));
+           output.z = LERP(output.z, a1, (count == 32));
+           output.x = LERP(output.x, a0, (count == 0));
+           output.y = LERP(output.y, a1, (count == 0));
+           output.z = LERP(output.z, a2, (count == 0));
+           
+           return output;
+        }
+        
+        
+        uint64_t packFloat64(uint32_t zSign, int zExp, uint32_t zFrac0, uint32_t zFrac1)
+        {
+           nbl::hlsl::uint32_t2 z;
+        
+           z.x = zSign + (uint32_t(zExp) << 20) + zFrac0;
+           z.y = zFrac1;
+           
+           uint64_t output = 0u;
+           output |= (uint64_t(z.x) << 32) & 0xFFFFFFFF00000000ull;
+           output |= uint64_t(z.y);
+           return  output;
+        }
+
+        
+        uint64_t roundAndPackFloat64(uint32_t zSign, int zExp, uint32_t zFrac0, uint32_t zFrac1, uint32_t zFrac2)
+        {
+           bool roundNearestEven;
+           bool increment;
+        
+           roundNearestEven = true;
+           increment = int(zFrac2) < 0;

```

Comment:
after everything is tested and implemented, try to optimize/rewrite this function, it seems to do a lot of complicated things because it can't treat the fractions as a single `uint64_t`